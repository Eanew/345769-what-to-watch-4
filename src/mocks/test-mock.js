import {toKebabCase} from "../utils/common.js";
import {APPROVED_GENRES} from "../utils/const.js";

const filmsTitles = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`];

const usersNames = [
  `Viktor Savinov`,
  `Alexandra Pushina`,
  `Julia Kim`,
  `Anton Timoshenkov`];

export const promo = {
  filmTitle: filmsTitles[0],
  genre: APPROVED_GENRES[0],
  release: 2014,
};

export const film = {
  id: filmsTitles[0] + 1,
  filmTitle: filmsTitles[0],
  release: 2011,
  runtime: 143,
  genre: APPROVED_GENRES[0],

  rating: {
    value: 6.3,
    votesCount: 2134,
  },

  image: {
    preview: `img/${toKebabCase(filmsTitles[0])}.jpg`,
    background: `img/bg-${toKebabCase(filmsTitles[0])}.jpg`,
    poster: `img/${toKebabCase(filmsTitles[0])}-poster.jpg`,
  },

  movie: {
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    full: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },

  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,

  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  isFavorite: false,
};

export const films = filmsTitles.map((title, i) => ({
  id: title + i,
  filmTitle: title,
  release: 2011 + i,
  runtime: 30 + i * 60,
  genre: APPROVED_GENRES[i],

  rating: {
    value: 3.4 + i,
    votesCount: 153 + i * 421,
  },

  image: {
    preview: `img/${toKebabCase(title)}.jpg`,
    background: `img/bg-${toKebabCase(title)}.jpg`,
    poster: `img/${toKebabCase(title)}-poster.jpg`,
  },

  movie: {
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    full: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },

  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,

  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  isFavorite: i % 2,
}));

export const review = {
  id: `comment1`,
  userName: usersNames[0],
  rating: 6.3,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`,
};

export const reviews = usersNames.map((name, i) => ({
  id: `comment${i + 1}`,
  userName: name,
  rating: 3.2 + i,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`,
}));
