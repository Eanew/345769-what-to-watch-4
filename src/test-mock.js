import {toKebabCase} from "./utils/common.js";
import {Genre, Screen, MovieTab} from "./utils/const.js";
import NameSpace from "./reducer/name-space.js";

const filmsTitles = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`];

const usersNames = [
  `Viktor Savinov`,
  `Alexandra Pushina`,
  `Julia Kim`,
  `Anton Timoshenkov`];

export const userInfo = {
  isAuthorized: true,
  id: 1,
  name: usersNames[2],
  email: `julia.kim@gmail.com`,
  avatar: `img/avatar.jpg`,
};

export const film = {
  id: filmsTitles[0] + 1,
  filmTitle: filmsTitles[0],
  release: 2011,
  runtime: 143,
  genre: Genre.SCI_FI,

  rating: {
    value: 6.3,
    votesCount: 2134,
  },

  image: {
    preview: `img/${toKebabCase(filmsTitles[0])}.jpg`,
    poster: `img/${toKebabCase(filmsTitles[0])}-poster.jpg`,
    background: `img/bg-${toKebabCase(filmsTitles[0])}.jpg`,
    backgroundColor: `#ffffff`,
  },

  movie: {
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    full: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },

  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,

  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  isFavorite: false,
};

export const films = filmsTitles.map((title, i) => ({
  id: title + i,
  filmTitle: title,
  release: 2011 + i,
  runtime: 30 + i * 60,
  genre: Object.values(Genre)[i],

  rating: {
    value: 3.4 + i,
    votesCount: 153 + i * 421,
  },

  image: {
    preview: `img/${toKebabCase(title)}.jpg`,
    poster: `img/${toKebabCase(title)}-poster.jpg`,
    background: `img/bg-${toKebabCase(title)}.jpg`,
    backgroundColor: `#ffffff`,
  },

  movie: {
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    full: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },

  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,

  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  isFavorite: (i % 2) ? true : false,
}));

export const review = {
  id: `comment1`,
  userId: 1,
  userName: usersNames[0],
  rating: 6.3,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`,
};

export const reviews = usersNames.map((name, i) => ({
  id: i + 1,
  userId: (i + 1) * 3,
  userName: name,
  rating: 3.2 + i,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`,
}));

export const mockStore = {
  [NameSpace.DATA]: {
    promo: films[0],
    films,
    favorites: films.filter((movie) => movie.isFavorite),
    reviews,
  },

  [NameSpace.USER]: {
    userInfo,
  },

  [NameSpace.FILMS]: {
    genre: Genre.ALL,
    displayedFilmsCount: 8,
  },

  [NameSpace.SCREEN]: {
    screen: Screen.MAIN,
    lastScreen: Screen.MAIN,
    movieTab: MovieTab.OVERVIEW,
  },
};
