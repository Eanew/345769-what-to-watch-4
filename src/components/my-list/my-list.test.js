import React from "react";
import renderer from "react-test-renderer";
import MyList from "./my-list.jsx";
import {Switch, Route, Router} from "react-router-dom";
import history from "../../history.js";

import {userInfo, films} from "../../test-mock.js";

it(`Should MyListComponent render correctly`, () => {
  const tree = renderer.create(
      <Router history={history}>
        <Switch>
          <Route exact path="">
            <MyList
              userInfo={userInfo}
              films={films}
              onFilmCardClick={() => {}}
              onLogoLinkClick={() => {}}
            />
          </Route>
        </Switch>
      </Router>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
