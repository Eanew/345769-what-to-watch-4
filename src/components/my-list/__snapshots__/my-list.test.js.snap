// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MyListComponent render correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="img/avatar.jpg"
          width="63"
        />
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            controls={false}
            height="175"
            loop={true}
            muted={true}
            poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
            preload="none"
            width="280"
          >
            <source
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              type="video/mp4"
            />
          </video>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href=""
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            controls={false}
            height="175"
            loop={true}
            muted={true}
            poster="img/bohemian-rhapsody.jpg"
            preload="none"
            width="280"
          >
            <source
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              type="video/mp4"
            />
          </video>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href=""
          >
            Bohemian Rhapsody
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            controls={false}
            height="175"
            loop={true}
            muted={true}
            poster="img/macbeth.jpg"
            preload="none"
            width="280"
          >
            <source
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              type="video/mp4"
            />
          </video>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href=""
          >
            Macbeth
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
